relayer/relays/beacon/header/syncer/syncer.go

```diff
@@ -246,13 +411,73 @@ func (s *Syncer) GetHeaderUpdate(blockRoot common.Hash) (HeaderUpdate, error) {
 		return HeaderUpdate{}, fmt.Errorf("convert block to scale: %w", err)
 	}
 
+	proofScale, err := s.getBlockHeaderAncestryProof(int(blockScale.Slot))
+
 	headerUpdate := HeaderUpdate{
-		Block: blockScale,
+		Block:          blockScale,
+		BlockRootProof: proofScale,
 	}
 
 	return headerUpdate, nil
 }
 
+func (s *Syncer) getBlockHeaderAncestryProof(slot int) ([]types.H256, error) {
+	var blockRootsContainer state.BlockRootsContainer
+
+	if s.activeSpec == config.Minimal {
+		blockRootsContainer = &state.BlockRootsContainerMinimal{}
+	} else {
+		blockRootsContainer = &state.BlockRootsContainerMainnet{}
+	}
+
+	blockRootsContainer.SetBlockRoots(s.currentFinalizedHeader.blockRoots)
+
+	indexInArray := slot % s.MaxSlotsPerHistoricalRoot
+
+	tree, err := blockRootsContainer.GetTree() // TODO check memory cost to save this in memory instead of block root array
+	if err != nil {
+		return nil, fmt.Errorf("convert block roots to tree: %w", err)
+	}
+
+	log.WithField("tree root", common.BytesToHash(tree.Hash())).Info("tree root")
+
+	treeDepth := math.Floor(math.Log2(float64(s.MaxSlotsPerHistoricalRoot)))
+	leavesStartIndex := int(math.Pow(2, treeDepth))
+	leafIndex := leavesStartIndex + indexInArray
```

Comment:
just curious why we add an extra `leavesStartIndex` for the leafIndex?

Reply:
The tree root leaves are the block_root values. For example, using this diagram: https://github.com/protolambda/eth2-docs#ssz-hash-tree-root-and-merkleization

`block_root[0]` would be located at tree index `8` in the example tree. So if you want to prove `block_root` at index `2` (`block_root[2]`), you would have to calculate `8` + `2` = tree index `10`. Hope this makes sense. If you have a better way to calculate it I'd be happy to change it.

Reply:
I see, the base index of `block_root` in original state tree, right?

I would assume `leavesStartIndex` is a const value for mainet spec or mimal spec. so maybe we can precaculate as a const in runtime without the exp/log functions?

Reply:
That's a good point. I added the config in the parachain but not in the relayer yet, will do so.

Reply:
I've cleaned this up: https://github.com/Snowfork/snowbridge/pull/765/files#diff-c589b232c311e67a8690c53fe1886c463a185178fde8ed4aa9dd2def499603b3R497